### PR TITLE
Add "always" and "never" options to "one-var" rule

### PR DESCRIPTION
**Pull request by dannyfritz**
As per #1619

### CODE REVIEW

File: lib/rules/one-var.js [hunk 1]
```diff
@@ -26,7 +28,7 @@ module.exports = function(context) {
         functionStack.pop();
     }
 
-    function checkDeclarations(node) {
+    function oneDeclaration(node) {
         if (functionStack[functionStack.length - 1]) {
```
Comment (nzakas) [inline lib/rules/one-var.js]: Function names should begin with a verb

File: lib/rules/one-var.js [hunk 2]
```diff
@@ -34,6 +36,28 @@ module.exports = function(context) {
         }
     }
 
+    function onlyUndefinedDeclarations(node) {
```
Comment (nzakas) [inline lib/rules/one-var.js]: Please add JSDoc comments for each function declaration.

File: docs/rules/one-var.md [hunk 3]
```diff
@@ -48,6 +50,63 @@ function foo() {
...
+function foo() {
+    var bar;
+
+    if (baz) {
+        var qux = true;
+    }
+}
+```
+
+When configured with `"only-undefined"` as the first option, the following patterns are considered warnings:
```
Comment (nzakas) [inline docs/rules/one-var.md]: I dont think "only-undefined" makes sense, please remove.

**Comment by nzakas** [comment]
Not a bad start, please read over our contribution guidelines to see how PRs must be submitted: http://eslint.org/docs/developer-guide/contributing.html

**Comment by dannyfritz** [comment]
I'll reopen a new one that follows the guidelines.

**Pull request closed without merging**